quarkus/tests/junit5/src/main/java/org/keycloak/it/utils/DockerKeycloakDistribution.java

```diff
@@ -23,10 +24,26 @@
 import java.util.Map;
 import java.util.concurrent.Executor;
 import java.util.concurrent.Executors;
+import java.util.function.Consumer;
 import java.util.stream.IntStream;
 
 public final class DockerKeycloakDistribution implements KeycloakDistribution {
 
+    private static class BackupConsumer implements Consumer<OutputFrame> {
+
+        final ToStringConsumer stdOut = new ToStringConsumer();
+        final ToStringConsumer stdErr = new ToStringConsumer();
+
+        @Override
+        public void accept(OutputFrame t) {
+            if (t.getType() == OutputType.STDERR) {
+                stdErr.accept(t);
+            } else if (t.getType() == OutputType.STDOUT) {
+                stdOut.accept(t);
+            }
+        }
+    }
```

Comment:
This is quite similar to https://github.com/keycloak/keycloak/pull/34956/files#diff-6b0c5824a800a0027bae134fa2edbeb32a6575a5bbe3addd033a9c1a5c48787a. Maybe we can try and align it even though it's not merged yet.

Reply:
> Maybe we can try and align it even though it's not merged yet.

We shouldn't try to do that proactively. It should be straight-forward later to have this implement OutputConsumer if the docker dist needs the same abstraction.